src/transformers/modeling_utils.py

```diff
@@ -2093,7 +2094,35 @@ def save_pretrained(
 
         # Save the model
         if state_dict is None:
-            state_dict = model_to_save.state_dict()
+            # if model parameters are offloaded, onload and send state dicts to CPU
+            if hasattr(model_to_save, "_hf_hook") and isinstance(model_to_save._hf_hook, AlignDevicesHook):
```

Comment:
Let's add another condition so that we only go through this logic when part of the model is offloaded. You can do that by checking if any param is on the meta device or not.

Reply:
Nice suggestion! I have included this check in the same if clause.

Reply:
Seems like we still go through this code when we are in a multi-gpu setup. Just checking if the model has hooks is not enough. One way to check for offloaded modules is if there is any module that has `module._hf_hook.offload`.

Reply:
I think I see what you mean and will change this so the if condition only activates for models with offloaded params